File: src/claude_memory_kit/tools/auto_extract.py
import logging

from ..config import get_api_key
from ..extract import extract_memories
from ..store import Store
from .remember import do_remember

log = logging.getLogger("cmk")


async def do_auto_extract(
    store: Store, transcript: str, user_id: str = "local"
) -> str:
    api_key = get_api_key()
    if not api_key:
        return "No API key configured. Cannot extract memories."

    memories = await extract_memories(transcript, api_key)
    if not memories:
        return "No memories worth keeping from this transcript."

    saved = []
    for mem in memories:
        try:
            msg = await do_remember(
                store,
                mem.get("content", ""),
                mem.get("gate", "epistemic"),
                mem.get("person"),
                mem.get("project"),
                user_id=user_id,
            )
            saved.append(msg)
        except Exception as e:
            log.warning("auto-extract save failed: %s", e)

    return (
        f"Auto-extracted {len(saved)} memories from transcript:\n"
        + "\n".join(saved)
    )

File: src/tools/auto_extract.rs
use anyhow::Result;
use crate::server::MemoryServer;
use crate::extract;

impl MemoryServer {
    pub async fn do_auto_extract(&self, transcript: &str) -> Result<String> {
        let memories = extract::extract_memories(transcript, &self.api_key).await?;

        if memories.is_empty() {
            return Ok("No memories worth keeping from this transcript.".into());
        }

        let mut saved = Vec::new();
        for mem in &memories {
            match self.do_remember(
                &mem.content,
                &mem.gate,
                mem.person.as_deref(),
                mem.project.as_deref(),
            ).await {
                Ok(msg) => saved.push(msg),
                Err(e) => {
                    tracing::warn!("auto-extract save failed: {}", e);
                }
            }
        }

        Ok(format!(
            "Auto-extracted {} memories from transcript:\n{}",
            saved.len(),
            saved.join("\n")
        ))
    }
}
